src/test/groovy/org/prebid/server/functional/tests/storage/AccountS3Spec.groovy

```diff
@@ -0,0 +1,118 @@
+package org.prebid.server.functional.tests.storage
+
+import org.prebid.server.functional.model.AccountStatus
+import org.prebid.server.functional.model.config.AccountConfig
+import org.prebid.server.functional.model.request.auction.BidRequest
+import org.prebid.server.functional.service.PrebidServerException
+import org.prebid.server.functional.service.PrebidServerService
+import org.prebid.server.functional.service.S3Service
+import org.prebid.server.functional.testcontainers.PbsServiceFactory
+import org.prebid.server.functional.util.PBSUtils
+
+import static io.netty.handler.codec.http.HttpResponseStatus.UNAUTHORIZED
+
+class AccountS3Spec extends StorageBaseSpec {
+
+    protected PrebidServerService s3StorageAccountPbsService = PbsServiceFactory.getService(s3StorageConfig +
+                                                                                            mySqlDisabledConfig +
+                                                                                            ['settings.enforce-valid-account': 'true'])
+
+    def "PBS should process request when active account is present in S3 storage"() {
+        given: "Default BidRequest with account"
+        def accountId = PBSUtils.randomNumber as String
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            setAccountId(accountId)
+        }
+
+        and: "Active account config"
+        def account = new AccountConfig(id: accountId, status: AccountStatus.ACTIVE)
+
+        and: "Saved account in AWS S3 storage"
+        S3_SERVICE.uploadAccount(DEFAULT_BUCKET, account)
+
+        when: "PBS processes auction request"
+        def response = s3StorageAccountPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should contain seatbid"
+        assert response.seatbid.size() == 1
+    }
+
+    def "PBS should throw exception when inactive account is present in S3 storage"() {
+        given: "Default BidRequest with account"
+        def accountId = PBSUtils.randomNumber as String
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            setAccountId(accountId)
+        }
+
+        and: "Inactive account config"
+        def account = new AccountConfig(id: accountId, status: AccountStatus.INACTIVE)
+
+        and: "Saved account in AWS S3 storage"
+        S3_SERVICE.uploadAccount(DEFAULT_BUCKET, account)
+
+        when: "PBS processes auction request"
+        s3StorageAccountPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS should reject the entire auction"
+        def exception = thrown(PrebidServerException)
+        assert exception.statusCode == UNAUTHORIZED.code()
+        assert exception.responseBody == "Account $accountId is inactive"
+    }
+
+    def "PBS should throw exception when account have different id inside json in S3 storage"() {
```

Comment:
`PBS should throw exception when account id is't match wiht bid request account id`
Please check other similar case

Reply:
done